utils/s2n_init.c

```diff
@@ -24,20 +24,34 @@
 #include "utils/s2n_mem.h"
 #include "utils/s2n_random.h"
 #include "utils/s2n_safety.h"
+#include "utils/s2n_safety_macros.h"
 
 #include "openssl/opensslv.h"
 
 #include "pq-crypto/s2n_pq.h"
 
+#include <pthread.h>
+
 static void s2n_cleanup_atexit(void);
 
 unsigned long s2n_get_openssl_version(void)
 {
     return OPENSSL_VERSION_NUMBER;
 }
 
+static pthread_t main_thread = 0;
+
+static bool atexit_cleanup = true;
+int s2n_disable_atexit(void) {
+    const bool already_initialized = (main_thread != 0);
```

Comment:
Can you make this comparison? The man page suggests it might not work: https://man7.org/linux/man-pages/man3/pthread_self.3.html You might need pthread_equal. And even then, can we guarantee 0 isn't a valid pthread_t?

Maybe use a separate variable for this.

Reply:
Good catch on the comparison! I had no idea

Reply:
I'm going to do a followup PR on this.